### PR TITLE
[ENG-4154][ENG-4162] Metadata forms

### PR DESCRIPTION
**Pull request by brianjgeiger**
Tickets: 
- [ENG-4154]
- [ENG-4162]

## Purpose

Add editable forms to the metadata pages. Note: No formatting or tests yet; those will come with a later ticket.

## Summary of Changes

1. Add adapters and serializers for new models
2. Hide edit buttons in edit mode
3. Add file metadata edit form
4. Add node metadata edit form (with the exception of the Funder widget)
5. Add language codes
6. Add resource type options
7. Update managers to handle language selection
8. Enable save for metadata forms
9. Fix mirage self link for custom-item-metadata-record
10. Add link to metadata page for registrations

## Screenshot(s)

Screen Shot 2022-11-29 at 4 12 20 PM
Screen Shot 2022-12-05 at 9 39 21 AM
Screen Shot 2022-12-05 at 9 38 52 AM

## Side Effects

None that I can think of.

## QA Notes

This affects project, registration, and file metadata pages.

[ENG-4154]: https://openscience.atlassian.net/browse/ENG-4154?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ENG-4162]: https://openscience.atlassian.net/browse/ENG-4162?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/osf-components/addon/components/node-metadata-manager/component.ts [hunk 1]
```diff
@@ -44,12 +76,25 @@ export default class NodeMetadataManagerComponent extends Component<Args> {
...
+        'isEditingDescription',
+        'isEditingFunding',
+        'isEditingResources',
+    ) inEditMode!: boolean;
+    @tracked isEditingDescription = false;
+    @tracked isEditingFunding = false;
+    @tracked isEditingResources = false;
     @tracked userCanEdit!: boolean;
-    @alias( 'getGuidMetadata.isRunning') isGatheringData!: boolean;
+    @or( 'getGuidMetadata.isRunning', 'cancel.isRunning') isGatheringData!: boolean;
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/node-metadata-manager/component.ts:89]: Note to self: this needs updating with new methods.

File: lib/osf-components/addon/components/node-metadata-manager/component.ts [hunk 2]
```diff
@@ -75,30 +120,104 @@ export default class NodeMetadataManagerComponent extends Component<Args> {
...
+        this.isEditingFunding = false;
+        this.isEditingResources = false;
+    }
+
+    @task
+    @waitFor
+    async cancelNode(){
+        if (this.saveErrored){
+            await this.node.reload();
+            this.saveErrored = false;
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/node-metadata-manager/component.ts:183]: Note to self: this.saveNodeErrored

File: lib/osf-components/addon/components/node-metadata-form/template.hbs [hunk 3]
```diff
@@ -1,46 +1,170 @@
...
+            {{t 'general.cancel'}}
+        </Button>
+        <Button
+            data-test-save-funding-metadata-button
+            aria-label={{t 'general.save'}}
+            data-analytics-name='Save funding metadata'
+            {{on 'click' @manager.saveMetadata}}
+        >
+            {{t 'general.save'}}
+        </Button>
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/node-metadata-form/template.hbs:121]: Note to self: disable save buttons when saving.

**Review by futa-ikeda — CHANGES_REQUESTED** [review]
A few minor things, but otherwise, looks great!

File: app/utils/languages.ts [hunk 4]
```diff
@@ -0,0 +1,1955 @@
+export const languageCodes = [
```
[the remark this answers is not in the document]
Comment (brianjgeiger) [reply app/utils/languages.ts:1]: This is an ISO standard, and not something we got from a codebase. There's no good way to make it a dependency, and it shouldn't update particularly frequently.

**Review by futa-ikeda — APPROVED** [review]

**Pull request merged**